src/lib/pipes/normalise-event-pipe.js

```diff
@@ -0,0 +1,82 @@
+import Pipe from './pipe.js';
+import Utils from '../utils.js';
+import NormalisedEvent from '../normalised-event.js';
+
+// Normalises an object with a type of Event
+class NormaliseEventPipe extends Pipe {
+  run(){
+
+    return new Promise(async resolve => {
+
+        let id = this.getId();
+        let type = this.getType();
+        let kind = this.getKind();
+        console.log(`Running ${id} (${type}) through ${this.constructor.name}`);
+
+        if (type == 'Event' || type == 'OnDemandEvent'){
+            // The top level event is the Event
+
+            this.doCleanup();
+
+            if (typeof this.rawData.superEvent !== 'undefined'){
+                // It has a superEvent which we can get more data from.
+                let {superEvent, ...event} = this.rawData;
```

Comment:
I don't understand what the ... syntax does on the left hand side of an assign statement?

Reply:
It assigns the superEvent property on the rawData object to a variable called superEvent, and all the rest of the object to the variable called event.

Reply:
ah ok